activerecord/lib/active_record/counter_cache.rb

```diff
@@ -180,7 +180,8 @@ def _create_record(*)
         each_counter_cached_associations do |association|
           if send(association.reflection.name)
             association.increment_counters
-            @_after_create_counter_called = true
+            @_after_create_counter_called ||= {}
```

Comment:
It doesn't assume anything.  It just wanted to set the flag to true and it didn't matter whether it was defined or not.  I just want to set the flag to true *for a key*, so I need to make sure the hash is defined.